presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/WindowOperatorStats.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.planPrinter;
+
+import com.facebook.presto.operator.WindowInfo;
+import com.facebook.presto.util.Mergeable;
+
+class WindowOperatorStats
+        implements Mergeable<WindowOperatorStats>
+{
+    private final int activeDrivers;
+    private final int totalDrivers;
+    private final double positionsInIndexesSumSquaredDiffs;
+    private final double sizeOfIndexesSumSquaredDiffs;
+    private final double indexCountPerDriverSumSquaredDiffs;
+    private final double partitionRowsSumSquaredDiffs;
+    private final double rowCountPerDriverSumSquaredDiffs;
+    private final long totalRowCount;
+    private final long totalIndexesCount;
+    private final long totalPartitionsCount;
+
+    public static WindowOperatorStats create(WindowInfo info)
+    {
+        int activeDrivers = 0;
+        int totalDrivers = 0;
+
+        double partitionRowsSumSquaredDiffs = 0.0;
+        double positionsInIndexesSumSquaredDiffs = 0.0;
+        double sizeOfIndexesSumSquaredDiffs = 0.0;
+        double indexCountPerDriverSumSquaredDiffs = 0.0;
+        double rowCountPerDriverSumSquaredDiffs = 0.0;
+        long totalRowCount = 0;
+        long totalIndexesCount = 0;
+        long totalPartitionsCount = 0;
+
+        double averageNumberOfIndexes = info.getWindowInfos().stream()
+                .filter(windowInfo -> windowInfo.getTotalRowsCount() > 0)
+                .mapToLong(windowInfo -> windowInfo.getNumberOfIndexes())
+                .average()
+                .orElse(Double.NaN);
+
+        double averageNumberOfRows = info.getWindowInfos().stream()
+                .filter(windowInfo -> windowInfo.getTotalRowsCount() > 0)
+                .mapToLong(windowInfo -> windowInfo.getTotalRowsCount())
+                .average()
+                .orElse(Double.NaN);
+
+        for (WindowInfo.DriverWindowInfo driverWindowInfo : info.getWindowInfos()) {
+            long partitionRowsCount = driverWindowInfo.getTotalRowsCount();
+            totalDrivers++;
+            if (partitionRowsCount > 0) {
+                long numberOfIndexes = driverWindowInfo.getNumberOfIndexes();
+                long rowCount = driverWindowInfo.getTotalRowsCount();
+
+                partitionRowsSumSquaredDiffs += driverWindowInfo.getSumSquaredDifferencesSizeInPartition();
```

Comment:
`partitionRowsSumSquaredDiffs` is used to determine std dev of rows in partitions, so with the assumption that every `WindowOperator` must have at least one partition (and always full partitions, never fractions), it shouldn't matter how many rows it processed per driver, to properly answer the question "how dispersed are rows among partitions" IMO, correct me if I'm wrong, though.